core/examples/http_bench_json_ops.rs

```diff
@@ -42,39 +42,35 @@ impl log::Log for Logger {
 }
 
 fn create_isolate() -> JsRuntime {
-  let state = BasicState::new();
-  state.register_op_json_sync("listen", op_listen);
-  state.register_op_json_sync("close", op_close);
-  state.register_op_json_async("accept", op_accept);
-  state.register_op_json_async("read", op_read);
-  state.register_op_json_async("write", op_write);
-
   let startup_data = StartupData::Script(Script {
     source: include_str!("http_bench_json_ops.js"),
     filename: "http_bench_json_ops.js",
   });
-
-  JsRuntime::new(state, startup_data, false)
+  let mut runtime = JsRuntime::new(startup_data, false);
+  runtime.register_op_json_sync("listen", op_listen);
+  runtime.register_op_json_sync("close", op_close);
+  runtime.register_op_json_async("accept", op_accept);
+  runtime.register_op_json_async("read", op_read);
+  runtime.register_op_json_async("write", op_write);
+  runtime
 }
 
 fn op_listen(
-  state: &BasicState,
+  state: Rc<RefCell<OpState>>,
```

Comment:
For sync ops, I'd still suggest to pass a borrowed `OpState` instead of a `Rc<RefCell<_>>`.

The limited flexibility vs async ops is intentional: a sync op **should** be completely done when this function returns, and `state: &mut OpState` would make it impossible to accidentally "capture" the state.